azure/scope/managedcontrolplane.go

```diff
@@ -447,6 +447,9 @@ func (s *ManagedControlPlaneScope) ManagedClusterSpec(ctx context.Context) azure
 	if s.ControlPlane.Spec.LoadBalancerSKU != nil {
 		managedClusterSpec.LoadBalancerSKU = *s.ControlPlane.Spec.LoadBalancerSKU
 	}
+	if s.ControlPlane.Spec.OutboundType != nil {
+		managedClusterSpec.OutboundType = s.ControlPlane.Spec.OutboundType
+	}
```

Comment:
Since both of these fields are already pointers, we should be able to define this directly in the `managedClusterSpec` declaration above.